common/client.go

```diff
@@ -303,48 +373,102 @@ func (rc *RestClient) execMethod(method string, dest string, data interface{}, r
 	}
 	errStr := ""
 	if err != nil {
-		errStr = err.Error()
+		errStr = fmt.Sprintf("ERROR: <%v>", err)
 	}
-	log.Printf("\n\t=================================\n\t%s %s\n\t%s\n\t\n\t%s\n\t%s\n\t=================================", method, rc.url, reqBodyStr, bodyStr, errStr)
+	log.Printf("\n\t=================================\n\t%s %s: %d\n\t%s\n\n\t%s\n\t%s\n\t=================================", method, rc.url, resp.StatusCode, reqBodyStr, bodyStr, errStr)
 
 	if err != nil {
 		return err
 	}
 
-	if result == nil {
-		return nil
+	var unmarshalBodyErr error
+
+	//	TODO deal properly with 3xx
+	//	if resp.StatusCode >= 300 && resp.StatusCode < 400 {
+	//		log.Printf("3xx: %d %v", resp.StatusCode, resp.Header)
+	//	}
+	//
+	if result != nil {
+		if body == nil {
```

Comment:
?

Should we not just be able to write:

    if body != nil {
        unmarshal....
    }